src/test/java/seedu/address/logic/commands/CountryFilterCommandTest.java

```diff
@@ -1,5 +1,105 @@
 package seedu.address.logic.commands;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static seedu.address.testutil.TypicalClients.getTypicalAddressBook;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.client.Client;
+import seedu.address.model.client.ClientCountryMatchesInputCountryPredicate;
+import seedu.address.model.country.Country;
+
 public class CountryFilterCommandTest {
-    //TODO: fill in when Country field is added to Clients
+
+    private static final String[] COUNTRY_CODES = Locale.getISOCountries();
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validCountries_matchesExpectedFilterResult() {
+        try {
+            List<Client> initialClientList = new ArrayList<>();
+
+            for (Client client : model.getFilteredClientList()) {
+                initialClientList.add(client);
+            }
```

Comment:
For this, it would be more concise to just use the overloaded constructor of ArrayList `ArrayList​(Collection<? extends E> c)` to make the copy of the filtered client list

Either that or just initialClientList.addAll()

Reply:
Ok done